platform/bukkit/src/main/java/net/kyori/adventure/platform/bukkit/BukkitAudienceFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of adventure-platform, licensed under the MIT License.
+ *
+ * Copyright (c) 2018-2020 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.platform.bukkit;
+
+import net.kyori.adventure.audience.Audience;
+import net.kyori.adventure.platform.AudienceFactory;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.Plugin;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * A factory for getting and creating {@link Audience}s for the Bukkit API.
+ */
+public interface BukkitAudienceFactory extends AudienceFactory {
+    /**
+     * Creates a {@link BukkitAudienceFactory} for the given plugin.
+     *
+     * @param plugin the plugin
+     * @return the audience factory
+     */
+    static @NonNull BukkitAudienceFactory create(final @NonNull Plugin plugin) {
+        return BukkitPlatform.getInstance(plugin);
+    }
+
+    /**
+     * Gets an audience for an individual player.
+     *
+     * <p>If the player is not online, messages are silently dropped.</p>
+     *
+     * @param player a player
+     * @return a player audience
+     */
+    @NonNull Audience player(final @NonNull Player player);
```

Comment:
If we have command sender, do we need an explicit interface for players?

Reply:
This is just a mirror of the generic `player(UUID)` method from the parent class.

Reply:
I also thought about renaming `audience(CommandSender)` to `sender(CommandSender)`, to be more in-line with the other methods in `AudienceFactory`.

Wasn't so sure though.